src/models/installation.ts

```diff
@@ -1,22 +1,37 @@
-import { BOOLEAN, DataTypes, DATE, Model } from "sequelize";
+import { BOOLEAN, DataTypes, DATE } from "sequelize";
 import { Subscription } from "./subscription";
 import { encrypted, getHashedKey, sequelize } from "models/sequelize";
+import { EncryptedModel } from "models/encrypted-model";
+import { EncryptionSecretKeyEnum } from "utils/encryption-client";
 
 // TODO: this should not be there.  Should only check once a function is called
 if (!process.env.STORAGE_SECRET) {
 	throw new Error("STORAGE_SECRET is not defined.");
 }
 
-export class Installation extends Model {
+export class Installation extends EncryptedModel {
 	id: number;
 	jiraHost: string;
 	secrets: string;
 	sharedSecret: string;
+	safeSharedSecret: string; //LOL
```

Comment:
1. For gradually rollout / migration purpose, favor having a new column instead of reuse existing
2. Like in `1`, favor explicit new property name instead of some magic using same prop name but different db column etc.

Reply:
I'd recommend we use `encryptedSharedSecret` for both the field and column to follow convention .  After we move all shared secrets to be encrypted by cryptor, we can change the field name to be `sharedSecrets` again without changing the references :)

As for the migration, I think we should create an admin API that migrates things over so we have control over when things are transferred over.